detectable/src/main/java/com/synopsys/integration/detectable/detectables/pip/poetry/PoetryDetectable.java

```diff
@@ -60,6 +68,13 @@ public DetectableResult applicable() {
         if (poetryLock == null && pyprojectToml == null) {
             return new FilesNotFoundDetectableResult(PYPROJECT_TOML_FILE_NAME, POETRY_LOCK);
         }
+
+        PoetrySectionResult poetrySectionResult = poetrySectionParser.parseToolPoetrySection(pyprojectToml);
```

Comment:
I think keeping the PoetrySectionResult as the field could make this a little cleaner. That way here you don't have to do this second assignment. When you call the extract you can do the orElse(null) bit there, though arguable if it's better to do that there or here, removing this assignment seems to outweigh it in my opinion.